Don't update counters twice if non-slotted counter

danielwestendorf opened the pull request:
<!--
  First of all, thanks for contributing!

  If it's a typo fix or minor documentation update feel free to skip the rest of this template!
-->

## What is the purpose of this pull request?
Fix compatibility with non-slotted counter cache columns

Fixes #9


## What changes did you make? (overview)
Don't double increment when a counter column is not registered as a slotted counter

## Is there anything you'd like reviewers to focus on?

## Checklist

- [X] I've added tests for this change
- [ ] I've added a Changelog entry
- [ ] I've updated a documentation

Envek commented on lib/activerecord_slotted_counters/has_slotted_counter.rb:
```diff
@@ -182,9 +182,11 @@ def prepare_slotted_counters_params(ids, counters)
-      super unless self.class.registered_slotted_counter? attribute
-
-      self.class.update_counters(id, attribute => by, :touch => touch)
+      if self.class.registered_slotted_counter? attribute
+        self.class.update_counters(id, attribute => by, :touch => touch)
+      else
+        super
+      end
```
Purely stylistic (and optional) suggestion, but IMO guard clauses can be more readable and diff can be shorter if we just add `return` before call to `super` under condition:

```suggestion
      return super unless self.class.registered_slotted_counter? attribute

      self.class.update_counters(id, attribute => by, :touch => touch)
```

Envek reviewed (changes requested):
Thanks a lot for the pull request with the fix!

You have checked “I've added tests for this change” checkbox, but I can't see any test changes in this pull request, perhaps you forgot to commit it?

And don't hesitate to add a change log entry praising you :wink:

Also optional nitpick below ↓

danielwestendorf commented:
Sorry, I was in a hurry and just threw the PR up quickly. I've revised.

+ Fixed test which was providing a positive results
+ Updated style as suggested
+ Added changelog

Thanks for the gem!

palkan commented:
Thanks!

Released in 0.1.1
